src/main/scala/scalaz/stream/async/mutable/Queue.scala

```diff
@@ -45,10 +45,24 @@ trait Queue[A] {
    *
    * Please use `Topic` instead of `Queue` when all subscribers
    * need to see each value enqueued.
+   *
+   * This process is equivalent to `dequeueBatch(1)`.
    */
   def dequeue: Process[Task, A]
 
   /**
+   * Provides a process that dequeues in chunks.  Whenever *n* elements
+   * are available in the queue, `min(n, limit)` elements will be dequeud
+   * and produced as a single `Seq`.  Note that this naturally does not
+   * *guarantee* that `limit` items are returned in every chunk.  If only
+   * one element is available, that one element will be returned in its own
+   * sequence.  This method basically just allows a consumer to "catch up"
+   * to a rapidly filling queue in the case where some sort of batching logic
+   * is applicable.
+   */
+  def dequeueBatch(limit: Int = 0): Process[Task, Seq[A]]
```

Comment:
I'm guessing `0` is a sentinel value that means "give me all values available"? Not a fan... how about just two separate functions:

```scala
def dequeueAvailable: Process[Task, Seq[A]]
def dequeueBatch(limit: Int): Process[Task, Seq[A]]
```

If you want to implement these in terms of the same underlying primitive, that is fine.